src/Adapter/Product/QueryHandler/SearchProductsHandler.php

```diff
@@ -96,25 +90,29 @@ public function handle(SearchProducts $query): array
 
     /**
      * @param Product $product
+     * @param SearchProducts $query
      *
      * @return FoundProduct
      */
-    private function createFoundProductFromLegacy(Product $product): FoundProduct
+    private function createFoundProductFromLegacy(Product $product, SearchProducts $query): FoundProduct
     {
         //@todo: sort products alphabetically
 
         $priceTaxExcluded = Product::getPriceStatic($product->id, false);
+        $priceTaxIncluded = Product::getPriceStatic($product->id, true);
 
-        $foundProduct = new FoundProduct(
+        return new FoundProduct(
             $product->id,
             $product->name[$this->contextLangId],
-            $this->contextLocale->formatPrice($priceTaxExcluded, $this->contextCurrencyCode),
+            $this->contextLocale->formatPrice($priceTaxExcluded, $query->getAlphaIsoCode()->getValue()),
+            Tools::ps_round($priceTaxIncluded, 2),
```

Comment:
I don't think you should round values unless really necessary The display already manages this (and better since it uses the currency precision)

Reply:
Removed

Reply:
Removed